hack/lib/dev-env.sh

```diff
@@ -111,6 +118,22 @@ function write_authenticator_config() {
         "${authenticator_config_template}" > "${authenticator_config}"
 }
 
+function write_authenticator_with_dynamicfile_mode_config() {
```

Comment:
not huge fan of this _with_dynamicfile_mode_config / _with_dynamic_file pattern. Some possible alternatives :
- have write_authenticator_config take argument
- have write_authenticator_with_dynamicfile_mode_config call write_authenticator_config, and then do dynamicfile specific things (to reduce code duplication)